### PR TITLE
Quantity Bug Fix on Add Items View!

### PR DESCRIPTION
**Pull request by RossaMania**
## Description

When working on the Item Quantity functionality, changing the quantity on the Add Item view was one of the last things to be worked on, and was not given enough attention--after all, if somebody is going to add an item to their list they are going to want to add at least one item. After all, the user can always change the quantity later with that shiny Item Quantity Form component in the List Item view.

This pull request adds min and max attributes to the number input for the item quantity in the AddItemForm.tsx file. The min is set to 1, and the max is set to 100. This should keep the quantity between "1" and "100".

## Related Issue

This pull request fixes bugs found in the "As a user, I want to be able change the quantity or amount of an item once added to the list so I know how many of that specific item to buy!" issue's pull request.

## Acceptance Criteria

The acceptance criteria for this bug fix was to make a quantity of "0" or less than "0" an invalid input!

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

### Before

image

### After

image

### Enter A Number!
- Navigate to the Manage List page.
- Enter a number between 1 and 100 in the "How Many" text field!
- Enter an item name, and select an option for when to buy.
- Then click Submit.
You should see a confirmation toast that your item has been added!

### It's Spinning!
- Navigate to the Manage List page.
- Click the "spinner" down arrow down to "0". You can't! You should see a popover message instead!
- Toggle the "spinner" up arrow up to "100". You can't! You should see a popover message instead!
- Enter a number between 1 and 100 in the "How Many" text field!
- Enter an item name, and select an option for when to buy.
- Then click Submit.
You should see a confirmation toast that your item has been added!

### Backspace!
- Navigate to the Manage List page.
- Press down on that backspace or delete key to make the input blank! You can't have a blank input! You should see a popover message as if the input number is "0"!
- Enter a number between 1 and 100 in the "How Many" text field!
- Enter an item name, and select an option for when to buy.
- Then click Submit.
You should see a confirmation toast that your item has been added!

### CODE REVIEW

**Review by alex-andria — APPROVED** [review]
Nice job with this quick bug fix PR Ross! No longer able to add 0 items to the list 💪.

NIT: I noticed that the quantity number is stuck with 0 as the first integer in the field. Not a blocker but maybe something to smooth out for the user experience in the future.

Oct-06-2024 04-13-57

File: src/components/forms/AddItemForm.tsx
Comment (bbland1) [inline src/components/forms/AddItemForm.tsx:125]: praise: nice add and quick turn around on working on the bug!

**Review by bbland1 — APPROVED** [review]

**Comment by RossaMania** [comment]
> Nice job with this quick bug fix PR Ross! No longer able to add 0 items to the list 💪.
> 
> NIT: I noticed that the quantity number is stuck with 0 as the first integer in the field. Not a blocker but maybe something to smooth out for the user experience in the future.
> 
> 

I agree! Thank you for pointing that out.

Since the input is a number type, the minLength and maxLength attributes don't apply (since those are for strings).

I tried an if check for length, but that is unreachable since the input already checks for those min and max numbers.

I plan on researching this a bit further as I work on other issues.

**Comment by RossaMania** [comment]
Right now there are no merge conflicts. I am happy to wait until tomorrow morning to merge this request, so @eternalmaha and @zahrafalak get the chance to review this as well.

**Pull request merged**
